framework/src/modules/interoperability/utils.ts

```diff
@@ -345,61 +351,62 @@ export const checkInboxUpdateValidity = (
 		newInboxSize = size;
 		newInboxRoot = root;
 	}
-	// If inboxUpdate contains a non-empty messageWitness, then update newInboxRoot to the output
-	if (
-		!txParams.certificate.equals(EMPTY_BYTES) &&
-		txParams.inboxUpdate.messageWitness.partnerChainOutboxSize !== BigInt(0) &&
-		txParams.inboxUpdate.messageWitness.siblingHashes.length !== 0
-	) {
-		newInboxRoot = regularMerkleTree.calculateRootFromRightWitness(
-			newInboxSize,
-			newInboxAppendPath,
-			messageWitness.siblingHashes,
-		);
-	}
-
-	const proof = {
-		siblingHashes: outboxRootWitness.siblingHashes,
-		queries: [
-			{
-				key: partnerChainIDBuffer,
-				value: newInboxRoot as Buffer,
-				bitmap: outboxRootWitness.bitmap,
-			},
-		],
-	};
-	const outboxKey = rawStateStoreKey(STORE_PREFIX_OUTBOX_ROOT);
-	const querykeys = [outboxKey];
-	const isSMTRootValid = sparseMerkleTree.verify(
-		querykeys,
-		proof,
-		decodedCertificate.stateRoot,
-		SMT_KEY_LENGTH,
-	);
-	if (!isSMTRootValid) {
-		return {
-			status: VerifyStatus.FAIL,
-			error: new Error(
-				'Failed at verifying state root when messageWitness and certificate are non-empty.',
-			),
+	// non-empty certificate and an inboxUpdate
+	if (!isCertificateEmpty(decodedCertificate) && !isInboxUpdateEmpty(txParams.inboxUpdate)) {
+		// If inboxUpdate contains a non-empty messageWitness, then update newInboxRoot to the output
+		if (!isMessageWitnessEmpty(txParams.inboxUpdate.messageWitness)) {
+			newInboxRoot = regularMerkleTree.calculateRootFromRightWitness(
+				newInboxSize,
+				newInboxAppendPath,
+				messageWitness.siblingHashes,
+			);
+		}
+		const proof = {
+			siblingHashes: outboxRootWitness.siblingHashes,
+			queries: [
+				{
+					key: partnerChainIDBuffer,
```

Comment:
```
outboxKey = MODULE_ID_INTEROPERABILITY
            || STORE_PREFIX_OUTBOX
            || SHA-256(uint32be(partnerChainID)).
```
query key should be this? also for the `queryKeys` below

Reply:
for queryKeys, its already using `const outboxKey = rawStateStoreKey(STORE_PREFIX_OUTBOX_ROOT);`